common/OpTestSystem.py

```diff
@@ -618,3 +624,132 @@ def sys_execute_cmd_onto_bmc(self, i_cmd):
             return self.cv_BMC.sys_execute_cmd_onto_bmc(i_cmd)
         except OpTestError as e:
             return BMC_CONST.FW_FAILED
+
+
+    ###########################################################################
+    # IPMI Console interfaces
+    ###########################################################################
+
+    ##
+    # @brief It will get the ipmi sol console
+    #
+    # @return l_con object if success or raise OpTestError
+    #
+    def sys_get_ipmi_console(self):
+        self.l_con = self.cv_IPMI.ipmi_get_console()
+        return self.l_con
+
+    ##
+    # @brief This function make sure, ipmi console is activated and then login to the lpar if host is already up.
+    #
+    # @param i_con @type string: this is the console object used for ipmi sol console access and for lpar login.
+    #
+    # @return BMC_CONST.FW_SUCCESS or raise OpTestError
+    #
+    def sys_ipmi_lpar_login(self, i_con):
+        l_con = i_con
+        l_host = self.lpar_ip
+        l_user = self.lpar_user
+        l_pwd = self.lpar_passwd
+        l_rc = l_con.expect_exact(["[SOL Session operational.  Use ~? for help]\r\n", \
+        "Error: Unable to establish IPMI v2 / RMCP+ session", \
+        pexpect.TIMEOUT, pexpect.EOF], timeout=120)
+        if l_rc == 0:
+            print "Got ipmi console"
+        elif l_rc in [1,2,3]:
+            l_msg = "Error: not able to get IPMI console"
+            raise OpTestError(l_msg)
+
+        time.sleep(5)
+        l_con.send("\r")
+        time.sleep(3)
+        l_rc = l_con.expect_exact(["login: ", "#", "/ #", "Petitboot", pexpect.TIMEOUT, pexpect.EOF], timeout=120)
+        if l_rc == 0:
+            l_con.sendline(l_user)
+            l_rc = l_con.expect([r"[Pp]assword:", pexpect.TIMEOUT, pexpect.EOF], timeout=120)
+            time.sleep(5)
+            if l_rc == 0:
+                l_con.sendline(l_pwd)
+            else:
+                l_msg = "Error: lpar login failed"
+                raise OpTestError(l_msg)
+        elif l_rc in [2,3]:
```

Comment:
why not make these return codes "2" and "3" constants.
It would be helpful for the understanding what's going on here